src/main/scala/org/allenai/scholar/metrics/ErrorAnalysis.scala

```diff
@@ -0,0 +1,41 @@
+package org.allenai.scholar.metrics
+
+import java.time.Year
+
+import org.allenai.scholar._
+import org.allenai.scholar.metrics.PrecisionRecall._
+
+case class ErrorAnalysis(
+  metricName: String,
+  pr: PR,
+  details: Iterable[(String, Example[_])]
+)
+
+object ErrorAnalysis {
+  def computeMetrics[T](
+    truth: Map[String, T],
+    predicted: Map[String, T],
+    metrics: (String, T => Iterable[_])*
+  ) = {
+
+    val examples: Iterable[(String, (String, Example[_]))] =
+      for {
+        (id, trueData) <- truth.toList
```

Comment:
This is a gotcha where `toList` is actually needed.  Without it, `examples` will be a `Map` and tuples with the same String in the first slot will be thrown out.